### PR TITLE
mavlink: moved to ModuleParams

### PR DESCRIPTION
**Pull request by RomanBapst**
@dagar Here you go. Could you review, please?

### CODE REVIEW

File: src/modules/mavlink/mavlink_main.h [hunk 1]
```diff
@@ -621,19 +627,16 @@ class Mavlink
...
-	param_t			_param_broadcast;
-
-	unsigned		_system_type;
+	DEFINE_PARAMETERS(
+		(ParamInt<px4::params::MAV_SYS_ID>) _param_system_id,
+		(ParamInt<px4::params::MAV_COMP_ID>) _param_component_id,
+		(ParamInt<px4::params::MAV_PROTO_VER>) _param_mav_proto_version,
+		(ParamInt<px4::params::MAV_RADIO_ID>) _param_radio_id,
+		(ParamInt<px4::params::MAV_TYPE>) _param_system_type,
+		(ParamInt<px4::params::MAV_USEHILGPS>) _param_use_hil_gps,
```
Comment (dagar) [inline src/modules/mavlink/mavlink_main.h]: ParamBool

File: src/modules/mavlink/mavlink_main.h [hunk 2]
```diff
@@ -621,19 +627,16 @@ class Mavlink
...
+		(ParamInt<px4::params::MAV_FWDEXTSP>) _param_forward_externalsp,
```
Comment (dagar) [inline src/modules/mavlink/mavlink_main.h]: ParamBool

File: src/modules/mavlink/mavlink_main.h [hunk 3]
```diff
@@ -621,19 +627,16 @@ class Mavlink
...
+		(ParamInt<px4::params::MAV_BROADCAST>) _param_broadcast_mode
```
Comment (dagar) [inline src/modules/mavlink/mavlink_main.h]: ParamBool

File: src/modules/mavlink/mavlink_main.cpp [hunk 4]
```diff
@@ -2600,8 +2545,7 @@ void Mavlink::check_radio_config()
 		}
 
 		/* reset param and save */
-		_radio_id = 0;
-		param_set_no_notification(_param_radio_id, &_radio_id);
+		_param_radio_id.set(0);
```
Comment (dagar) [inline src/modules/mavlink/mavlink_main.cpp]: .set() -> commit_no_notification()

File: src/modules/mavlink/mavlink_main.h [hunk 5]
```diff
@@ -228,17 +229,17 @@ class Mavlink
 
 	bool			get_hil_enabled() { return _hil_enabled; }
 
-	bool			get_use_hil_gps() { return _use_hil_gps; }
+	bool			get_use_hil_gps() { return _param_use_hil_gps.get() > 0; }
```
Comment (dagar) [inline src/modules/mavlink/mavlink_main.h]: You can make these ParamBools and use them directly.

File: src/modules/mavlink/mavlink_main.h [hunk 6]
```diff
@@ -426,7 +427,15 @@ class Mavlink
 
 	ringbuffer::RingBuffer	*get_logbuffer() { return &_logbuffer; }
 
-	unsigned		get_system_type() { return _system_type; }
+	unsigned		get_system_type()
```
Comment (dagar) [inline src/modules/mavlink/mavlink_main.h]: Maybe leave it alone here print an error in the param update? You could also set the param to 0 there.

File: src/modules/mavlink/mavlink_main.h (hunk 2 above)
Comment (RomanBapst) [reply src/modules/mavlink/mavlink_main.h]: @dagar Ah cool, did not know about those.

**Comment by RomanBapst** [comment]
@dagar I addressed your comments.

**Comment by dagar** [comment]
CI failure is real. There's a test that exhaustively sets all params and then verifies them later. I believe mavlink could be removed from these entirely. https://github.com/PX4/Firmware/blob/master/posix-configs/SITL/init/test/tests_template.in#L24-L25

**Pull request merged**